core/src/main/scala/org/apache/spark/sql/crossdata/catalog/JDBCCatalog.scala

```diff
@@ -218,7 +218,10 @@ class JDBCCatalog(override val conf: CatalystConf = new SimpleCatalystConf(true)
       )
     }
     }
-    Some(StructType(structFields))
+    if(structFields.size == 0)
```

Comment:
I just got an new idea on how to simplify this conditional pattern [If a collection is empty return `None` and `Some(collection)`otherwise]: `structFields.headOption.map(structFields)`